### ISSUE TITLE
IIFE format does not expose named defaults when moduleName contains multiple parts

### ISSUE THREAD

**Issue by bdukes** [rollup/rollup#1492]
If you go to https://rollupjs.org/repl, select the _Named Exports_ example, switch the format to `iife`, and add a `.` to the module name (e.g. `myModule.subModule`), the `return exports;` statement at the end of the generated bundle is removed.

This is not an issue if there is only a `default export`, but shows up when named exports are in use (whether there is a `default export` or not).

An example is https://rollupjs.org/repl?version=0.45.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiUyRiUyMFRoZXJlJTIwYXJlJTIwbWFueSUyMHdheXMlMjB0byUyMGV4cG9ydCUyMGJpbmRpbmdzJTVDbiUyRiUyRiUyMGZyb20lMjBhbiUyMEVTMjAxNSUyMG1vZHVsZSU1Q25leHBvcnQlMjB2YXIlMjBmb28lMjAlM0QlMjAxJTNCJTVDbiU1Q25leHBvcnQlMjBmdW5jdGlvbiUyMGJhciUyMCgpJTIwJTdCJTVDbiU1Q3RyZXR1cm4lMjBmb28lM0IlMjAlMkYlMkYlMjB0cnklMjBjaGFuZ2luZyUyMHRoaXMlMjB0byUyMCU2MGZvbyUyQiUyQiU2MCU1Q24lN0QlNUNuJTVDbmZ1bmN0aW9uJTIwYmF6JTIwKCklMjAlN0IlNUNuJTVDdHJldHVybiUyMGJhcigpJTNCJTVDbiU3RCU1Q24lNUNuZXhwb3J0JTIwJTdCJTIwYmF6JTIwJTdEJTNCJTVDbmV4cG9ydCUyMColMjBmcm9tJTIwJy4lMkZxdXgnJTNCJTIyJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMnF1eC5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjB2YXIlMjBxdXglMjAlM0QlMjAnUVVYJyUzQiUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJpaWZlJTIyJTJDJTIybW9kdWxlTmFtZSUyMiUzQSUyMm15QnVuZGwuZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJtb2R1bGVJZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmV4YW1wbGUlMjIlM0ElMjIwMiUyMiU3RA==

**Comment by pshrmn** [comment]
This appears to be related to #1472 @Andarist . If you use the `extend` option, it should work correctly (but the REPL does not give you the opportunity to do this so you'll have to verify with a local project). I believe that there are a couple possible ways to approach fixing this:

1. Modify [this `if` statement](https://github.com/rollup/rollup/blob/v0.45.2/src/finalisers/iife.js#L33) to throw when using a name spaced `moduleName`, but not `extends`.
```js
if ( name && (possibleVariableAssignment && !isLegal(name)) || (isNamespaced && !extend) ) {
```
2. Make `extends` true by default when using a name spaced `moduleName`. Does that defeat the purpose of the option, though?
3. Modify `extends` to be the extended string instead of a boolean.
```js
{
  moduleName: 'myModule',
  extends 'subModule'
}
// or for even deeper
{
  moduleName: 'myModule',
  extends: 'one.two.three'
}
```

**Comment by Andarist** [comment]
@pshrmn thanks for the mention, ive created a fixing PR

**Comment by Rich-Harris** [comment]
released 0.46 with the fix — thanks

**Comment by Andarist** [comment]
Nothing to thank for, I'm the one who screwed it up in the first place 😄
